### DISCUSSION TITLE
How does the update equation changes if we set $g_a = 0$?

### DISCUSSION THREAD

**Discussion by Hailey0101** [qucontrol/krotov#114] [Q&A]
My question is more focused on theoretical aspects of the package. If I choose not to minimize the $g_a$ function, how does it impact the update equation (Eq.(12) in the documentation)? This update equation in Eq.(12) appears to have a strong dependence on Eq.(11) found on the same page, which represents the derivative of the $g_a$ function. So, I am wondering about the how the update equation of Eq.(12) should change when we disregard the significance of the $g_a$ function, specifically by setting $g_a = 0$ (so that a functional that we are trying to minimize is just $J_T$).

**Comment by goerz** [comment]
You cannot have Krotov's method without $g_a$. Don't think of the term $g_a = λ Δϵ^2 / S(t)$ as a "running cost" in the traditional sense as some physical quantity that you want to minimize. If you start from a gradient-descent (GRAPE) minimization of just a final-time functional $J_T$, then the equivalent optimization with Krotov's method would just add the $g_a$ term on a purely formal basis. If you want to assign meaning to it, think of it as "maximize convergence". That is, make the update $Δϵ$ smaller and smaller as the optimization progresses. That's also, in fact, why the parameters in that term ($λ_a$) control the convergence rate of the optimization.

But in reality, $g_a$ just a formal part of the derivation of Krotov's method, so don't worry about it too much. In fact, things get a bit tricky if you want to add other control-dependent running cost with a more physical meaning. It's possible, but only very specific forms of $g_a$ result in an update equation at all, and it's not supported in this implementation. In general, Krotov's method is best for unconstrained controls, so this shouldn't be that much of a problem.
